gradio/utils.py

```diff
@@ -1011,7 +1011,10 @@ def is_in_or_equal(path_1: str | Path, path_2: str | Path):
     """
     path_1, path_2 = abspath(path_1), abspath(path_2)
     try:
-        if ".." in str(path_1.relative_to(path_2)):  # prevent path traversal
+        path_1_without_filename = abspath("/".join(str(path_1).split("/")[:-1]))
```

Comment:
I think we should use `path_1.parent` here but not sure if there are any unintended security consequences of this change.

Reply:
Agree with @freddyaboulton. We should use `path1.parent` instead here. As long as `path1` is a file (which we should check for and raise if not), then I believe it should be fine

Reply:
@abidlabs @freddyaboulton Thanks for the suggestion. I have fixed this.